packages/neon-core/src/tx/main.ts

```diff
@@ -1,140 +1,134 @@
-  const version = parseInt(byte, 16);
-  if (version !== 0) {
-    log.error(`Transaction version should be 0 not ${version}`);
-  }
-  return Object.assign(tx, { version });
-}
-
-export function deserializeNonce(
-  ss: StringStream,
-  tx: Partial<TransactionLike> = {}
-): Partial<TransactionLike> {
-  const nonce = parseInt(ss.read(4), 16);
-  return Object.assign(tx, { nonce });
-}
-
-export function deserializeSender(
-  ss: StringStream,
-  tx: Partial<TransactionLike> = {}
-): Partial<TransactionLike> {
-  return Object.assign(tx, { sender: ss.read(20) });
-}
-
-export function deserializeScript(
-  ss: StringStream,
-  tx: Partial<TransactionLike> = {}
-): Partial<TransactionLike> {
-  const script = ss.readVarBytes();
-  if (script.length === 0) {
-    log.error("Script should not be vacant.");
-  }
-  return Object.assign(tx, { script });
-}
-
-export function deserializeFee(
-  ss: StringStream,
-  tx: Partial<TransactionLike> = {}
-): Partial<TransactionLike> {
-  const systemFee = fixed82num(ss.read(8));
-  const networkFee = fixed82num(ss.read(8));
-  return Object.assign(tx, { systemFee, networkFee });
-}
-
-export function deserializeValidUntilBlock(
-  ss: StringStream,
-  tx: Partial<TransactionLike>
-): Partial<TransactionLike> {
-  const validUntilBlock = parseInt(ss.read(4), 16);
-  return Object.assign(tx, { validUntilBlock });
-}
-
-export function getCosignersFromAttributes(
-  attrs: TransactionAttribute[] | TransactionAttributeLike[]
-): string[] {
-  return attrs
-    .filter(attr => toTxAttrUsage(attr.usage) === TxAttrUsage.Cosigner)
-    .map(attr => attr.data);
-}
-
-export function deserializeAttributes(
-  ss: StringStream,
-  tx: Partial<TransactionLike>
-): Partial<TransactionLike> {
-  const attributes = deserializeArrayOf(
-    TransactionAttribute.fromStream,
-    ss
-  ).map(i => i.export());
-  const cosigners = getCosignersFromAttributes(attributes);
-  if (
-    !cosigners.every(
-      cosigner =>
-        cosigners.indexOf(cosigner) === cosigners.lastIndexOf(cosigner)
-    )
-  ) {
-    log.error("Cosigner should not duplicate.");
-  }
-  return Object.assign(tx, { attributes });
-}
-
-export function deserializeWitnesses(
-  ss: StringStream,
-  tx: Partial<TransactionLike>
-): Partial<TransactionLike> {
-  const scripts = deserializeArrayOf(Witness.fromStream, ss).map(i =>
-    i.export()
-  );
-  return Object.assign(tx, { scripts });
-}
-
-export function formatSender(sender: string | undefined): string {
-  if (!sender) {
-    return "";
-  }
-  if (sender.length === 42 && sender.startsWith("0x")) {
-    const hex = sender.slice(2);
-    ensureHex(hex);
-    return hex;
-  } else if (sender.length === 40) {
-    ensureHex(sender);
-    return sender;
-  } else if (sender.length === 34) {
-    return getScriptHashFromAddress(sender);
-  } else {
-    throw new Error(`Sender format error: ${sender}`);
-  }
-}
+import { num2VarInt, StringStream, fixed82num, ensureHex } from "../u";
```

Comment:
Looks like a change in encoding please check.

Reply:
This is a renamed/relocated file. I guess this diff is what it suppose to be, cause what if they are renamed but the content is exactly the same? I tried to solve it but cannot.

Reply:
Fixed this.